### PR TITLE
Better error messages for PoA chains

### LINKED ISSUE #4030: Better account missing error
There is no difference between account missing and wrong/no password error message now with `--password`. Now with separate key directories it introduces confusion. If no account is present say something like:
"No account found for the current chain, please run `parity account new -d current-d --chain current-chain`"

### PR DESCRIPTION
**Pull request by maciejhirsz**
Closes #4030

### CODE REVIEW

File: parity/run.rs [hunk 1]
```diff
@@ -230,9 +230,21 @@ pub fn execute(cmd: RunCmd, can_restart: bool, logger: Arc<RotatingLogger>) -> R
 	miner.set_extra_data(cmd.miner_extras.extra_data);
 	miner.set_transactions_limit(cmd.miner_extras.transactions_limit);
 	let engine_signer = cmd.miner_extras.engine_signer;
+
 	if engine_signer != Default::default() {
+		// Check if engine signer exists
+		if !account_provider.has_account(engine_signer).unwrap_or(false) {
+			return Err(format!("Consensus signer account not found for the current chain, please run `parity account new -d current-d --chain current-chain`"));
```
Comment (rphmeier) [inline parity/run.rs]: Is this info available in the `RunCmd` somewhere? Would be nice to format it in for adaptable error messages.

**Comment by maciejhirsz** [comment]
[clabot:check]

File: parity/run.rs [hunk 2]
```diff
@@ -478,17 +490,27 @@ fn prepare_account_provider(dirs: &Directories, data_dir: &str, cfg: AccountsCon
...
+			return Err(format!("Account {} not found for the current chain, please run `parity account new -d current-d --chain current-chain --keys-path current-keys-path`", a));
+		}
+
+		// Check if any passwords have been read from the password file(s)
+		if passwords.is_empty() {
+			return Err(format!("No password found to unlock account {}. Make sure valid password is present in files passed using `--password` or in the configuration file.", a));
+		}
+
+		if !passwords.iter().any(|p| account_provider.unlock_account_permanently(a, (*p).clone()).is_ok()) {
+			return Err(format!("No valid password to unlock account {}. Make sure valid password is present in files passed using `--password` or in the configuration file.", a));
```
Comment (rphmeier) [inline parity/run.rs]: This (and other similar messages like on L#247) could be extracted to a constant.

Comment (maciejhirsz) [inline parity/run.rs]: Or abstract out the checks in both places to a function?

**Comment by maciejhirsz** [comment]
Updated according to suggestions from @keorn. Account creation hints are now adaptive:

```
$ parity --chain ./spec.json --config ./.parityrc -d ./dummy --unlock 0x1231222312312312321312313213231312323123
Loading config file from ./.parityrc
2017-01-04 17:03:13  Starting Parity/v1.5.0-unstable-9db3f38-20170103/x86_64-linux-gnu/rustc1.14.0
2017-01-04 17:03:13  State DB configuration: fast
2017-01-04 17:03:13  Operating mode: active
Account 1231…3123 not found for the current chain. You can create an account via RPC, UI or `parity account new --chain ./spec.json --keys-path ./dummy/keys`.
```

`keys-path` can be long (especially when it's a default, absolute path), but it can be copy-pasted safely.

**Pull request merged**